src/test/compile-fail/const-fn-error.rs

```diff
@@ -23,5 +23,5 @@ const fn f(x: usize) -> usize {
 
 #[allow(unused_variables)]
 fn main() {
-    let a : [i32; f(X)]; //~ NOTE for array length here
+    let a : [i32; f(X)]; //~ NOTE for constant expression here
```

Comment:
This is a slight decrease in diagnostic quality

Reply:
With moving to const generics the harcoded messages seem bizarre at best to me - the expression *happens* to be in an array length, the span is more important that that.
The description could potentially be recovered by walking up the HIR but that only works in the local crate.